PowerEditor/src/Notepad_plus.cpp

```diff
@@ -1351,46 +1351,29 @@ void Notepad_plus::removeEmptyLine(bool isBlankContained)
 	_findReplaceDlg.processAll(ProcessReplaceAll, &env, true);
 }
 
-void Notepad_plus::getMatchedFileNames(const TCHAR *dir, const vector<generic_string> & patterns, vector<generic_string> & fileNames, bool isRecursive, bool isInHiddenDir)
+void Notepad_plus::getMatchedFileNames(const TCHAR *dir, const vector<generic_string> & patterns, vector<generic_string> & fileNames, bool isRecursive, bool isInHiddenDir, const Progress* progress)

```

Comment:
When it comes to things passed by-value in C++ functions, there isn't too much value in making them const. Essentially you're trying to "protect the copy", which if changed, doesn't really break anything anyway. Also the change is not observable from the caller. You couldn't write a unit test to verify whether or not the copy was changed. This IMHO comes down to KISS principle; use const in more pragmatic ways and you eliminate some arguably unnecessary verbosity from your code and make it look more natural and readable.